lib/livebook/web_socket/server.ex

```diff
@@ -87,11 +87,25 @@ defmodule Livebook.WebSocket.Server do
     end
   end
 
+  @loop_ping_delay 5_000
+
   @impl true
+  def handle_info(:loop_ping, state) do
+    case Client.send(state.http_conn, state.websocket, state.ref, :ping) do
+      {:ok, conn, websocket} ->
+        Process.send_after(self(), :loop_ping, @loop_ping_delay)
+        {:noreply, %{state | http_conn: conn, websocket: websocket}}
+
+      {:error, conn, websocket, _reason} ->
+        {:noreply, %{state | http_conn: conn, websocket: websocket}}
+    end
+  end
+
   def handle_info(message, state) do
     case Client.receive(state.http_conn, state.ref, state.websocket, message) do
       {:ok, conn, websocket, :connected} ->
         state = send_received({:ok, :connected}, state)
+        send(self(), :loop_ping)
```

Comment:
You have to be careful because you can end-up with multiple ping loops. For example, if you disconnect and reconnect three times in a row, you will now have three `loop_ping`s going.

`Process.send_after` returns a reference. I think you need to store it on the state and then cancel it whenever the connection is cancelled.

Another approach, perhaps easier, is to generate a reference and include it in the message and store it in the state. Every time you connect, you generate a new reference. You only do the ping if the reference in the state and in the message match.